### ISSUE TITLE
[5.4] Eloquent object based events conflict with model observers

### ISSUE THREAD

**Issue by tillkruss** [laravel/framework#17885]
- Laravel Version: 5.4.11
- PHP Version: 7.1.0

### Description:
Eloquent object based events seem to conflict with model observers.

I'm using Scout's `Searchable` trait on a model, which registers an `updated` events internally. 

Now when I register my own `updated` event class [as described in the docs](https://laravel.com/docs/5.4/eloquent#events), the Scout `updated` event isn't being called any longer.

Is that intended behaviour, am I doing something wrong, or a bug?

**Comment by arjasco** [comment]
Interesting! I've just started using scout and also heavily rely on model observers.

Tried to replicate what you are experiencing but i have both events firing here. The events on my model get fired first before the observer as defined to do so [`here`](https://github.com/laravel/framework/blob/5.4/src/Illuminate/Database/Eloquent/Concerns/HasEvents.php#L139) 

If i return false from the handle method on the custom listener then thats the only time it will not propagate and get through to the observer.
